tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/TestingFrameworkEvpTest.cs

```diff
@@ -362,4 +412,232 @@ protected void InjectSession(
+            }
+        }
+        catch
+        {
+            Output.WriteLine("Framework Version: " + new Version(FrameworkDescription.Instance.ProductVersion));
+            if (!string.IsNullOrWhiteSpace(packageVersion))
+            {
+                Output.WriteLine("Package Version: " + new Version(packageVersion));
+            }
+
+            WriteSpans(executionData.Tests);
+            throw;
+        }
+    }
+
+    private static TValue? GetValueOrDefault<TKey, TValue>(IDictionary<TKey, TValue> dictionary, TKey key)
+        where TKey : notnull => dictionary.TryGetValue(key, out var value) ? value : default;
+
+    public readonly struct MockData
+    {
+        public readonly string SettingsJson;
+        public readonly string TestsJson;
+        public readonly string TestManagementTestsJson;
+
+        public MockData(string settingsJson, string testsJson, string testManagementTestsJson)
+        {
+            SettingsJson = settingsJson;
+            TestsJson = testsJson;
+            TestManagementTestsJson = testManagementTestsJson;
+        }
+
+        public override string ToString()
+        {
+            return $"SettingsJson: {SettingsJson}, TestsJson: {TestsJson}, TestManagementTestsJson: {TestManagementTestsJson}";
+        }
+    }
+
+    public readonly struct ExecutionData
+    {
+        public readonly List<MockCIVisibilityTest> Tests;
+        public readonly List<MockCIVisibilityTestSuite> TestSuites;
+        public readonly List<MockCIVisibilityTestModule> TestModules;
+
+        public ExecutionData()
+        {
+            Tests = new List<MockCIVisibilityTest>();
+            TestSuites = new List<MockCIVisibilityTestSuite>();
+            TestModules = new List<MockCIVisibilityTestModule>();
+        }
+
+        public delegate void ValidateDelegate(in ExecutionData data);
+    }
+
+    public readonly struct TestScenario
+    {
+        public readonly string? TypeName;
+        public readonly string FriendlyName;
+        public readonly MockData MockData;
+        public readonly int ExpectedExitCode;
+        public readonly int ExpectedSpans;
+        public readonly bool UseSnapshot;
+        public readonly bool UseDotnetExec;
+        public readonly ExecutionData.ValidateDelegate ValidateAction;
+
+        public TestScenario(string? typeName, string friendlyName, MockData mockData, int expectedExitCode, int expectedSpans, bool useSnapshot, ExecutionData.ValidateDelegate validateAction, bool useDotnetExec = true)
+        {
+            TypeName = typeName;
+            FriendlyName = friendlyName;
+            MockData = mockData;
+            ExpectedExitCode = expectedExitCode;
+            ExpectedSpans = expectedSpans;
+            UseSnapshot = useSnapshot;
+            UseDotnetExec = useDotnetExec;
+            ValidateAction = validateAction;
+        }
+    }
+
+    protected class MockLogsIntakeForCiVisibility : MockLogsIntake<MockLogsIntakeForCiVisibility.Log>
```

Comment:
Should this be more generic? can/should we move this support to the MockTracerAgent itself?

Reply:
`MockLogsIntake<T>` looks pretty generic to me already, I'm doing this as an alias with the custom `Log` format for CI Visibility. This is declared in `TestingFrameworkEvpTest` the abstract base class for all other testing frameworks tests.

Reply:
I mean we can move it, but I don't think that is it in a bad place right now.